test/System.Net.Http.Formatting.Test/HttpMessageContentTests.cs

```diff
@@ -46,20 +46,34 @@ private static HttpResponseMessage CreateResponse(bool containsEntity)
             return httpResponse;
         }
 
-        private static async Task<string> ReadContentAsync(HttpContent content)
+        private static async Task<string> ReadContentAsync(HttpContent content, bool unBuffered = false)
         {
-            await content.LoadIntoBufferAsync();
+            if (unBuffered)
+            {
+                var stream = new MemoryStream();
+                await content.CopyToAsync(stream);
+                stream.Position = 0L;
+
+                // StreamReader will dispose of the Stream.
+                using var reader = new StreamReader(stream);
```

Comment:
Why CopyToAsync vs ReadAsStreamAsync?
```suggestion
                // StreamReader will dispose of the Stream.
                using var reader = new StreamReader(await content.ReadAsStreamAsync());
```

Reply:
Good point.

nit: Since we're no longer allocating the `Stream`, the comment doesn't make that much sense anymore.
```suggestion
                using var reader = new StreamReader(await content.ReadAsStreamAsync());
```

Reply:
About to revert changes here. `ReadAsStreamAsync()` often returns a `LimitMemoryStream` and that can only be read once, invalidating many of the new tests in this PR.

Reply:
Oh, `ReadAsStreamAsync()` also buffers the `HttpContent`. That invalidates a few other tests and completely undermines the intent of the `unBuffered` argument to this test method.

Reply:
Fixed again

Reply:
ReadAsStreamAsync shouldn't buffer...?

Reply:
> ReadAsStreamAsync shouldn't buffer...?

Maybe it shouldn't but it does. See the default `HttpContent` implementations where `ReadAsStreamAsync()` ➡️ `CreateContentReadStreamAsync()` ➡️ `LoadIntoBufferAsync()` in <https://github.com/microsoft/referencesource/blob/master/System/net/System/Net/Http/HttpContent.cs> and <https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs>.

Reply:
Oh, you're supposed to override CreateContentReadStream/Async if you can produce a stream without buffering.

Reply:
This behaviour is what I saw in the tests; it's not about `HttpMessageContent`. Those tests mostly use `StreamContent` and I _think_ `StreamContent` itself overrides enough to avoid buffering. But one of the tests failed, clearly showing buffering. I don't remember which test unfortunately. (It got lost among the many failures due to the read-once behaviour.)

Reply:
I'm probably wrong here. `HttpMessageContent` overrides `SerializeToStreamAsync(Stream stream, TransportContext context)` but not `CreateContentReadStreamAsync()`. That may be why `ReadAsStreamAsync()` behaves badly. I can test an override but am unsure how signfiicant a bug this may be. Thoughts❔

Reply:
It's just in-efficient. I don't know that people will be calling ReadAsStreamAsync much. The WebApi infrastructure uses SerializeToStreamAsync.

Reply:
> It's just in-efficient. I don't know that people will be calling ReadAsStreamAsync much. The WebApi infrastructure uses SerializeToStreamAsync(?).

That's _somewhat_ true, I see 8 `content.CopyToAsync()` and 5 `content.ReadAsStreamAsync()` calls in the src/ of this repo. `SerializeToStreamAsync()` is the main `virtual` (actually `protected abstract`) method that `CopyToAsync()` uses; `ReadAsStreamAsync()` also uses `CreateContentReadStreamAsync()`. If you reconsider, feel free to file a bug in this repo.

My current perspective is that overriding `CreateContentReadStreamAsync()` would be difficult to achieve the desired "better way" documented e.g. at <https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs#L544>
and <https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs#L555-L557>
because there's no one `Stream` or `byte` array to wrap. Would also need to consider the other `HttpContent` derived classes in this repo, almost all of which do _not_ override `CreateContentReadStreamAsync()` as well as the fact this would be a breaking change.